src/client/components/LoginForm/LoginForm.tsx

```diff
@@ -1,56 +1,120 @@
-import React from 'react';
-import type { FunctionComponent } from 'react';
-
-const LoginForm: FunctionComponent = () => (
-  <form>
-    <div className="form-outline mb-4">
-      <input type="email" id="emailBox" className="form-control" />
-      <label className="form-label" htmlFor="emailBox">
-        Email address
-      </label>
-    </div>
-
-    <div className="form-outline mb-4">
-      <input type="password" id="passwordBox" className="form-control" />
-      <label className="form-label" htmlFor="passwordBox">
-        Password
-      </label>
-    </div>
-
-    <div className="row mb-4">
-      <div className="col">
+import React, { useState, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import type { FunctionComponent, ChangeEvent, SyntheticEvent } from 'react';
+import { isObjectRecord } from '../../common/utilities/types';
+
+const LoginForm: FunctionComponent = () => {
+  const navigate = useNavigate();
+  // check if user logged in already
+  const user = localStorage.getItem('user');
+  if (user !== null) {
+    navigate('/');
+  }
+
+  // const [error, setError] = useState<string | undefined>(undefined);
+  const [username, setUsername] = useState<string>('');
```

Comment:
Just a small note: We don't need to specify `<string>` here. TypeScript will infer the type based on the default empty string that you're passing in. Not a big deal, though. It's fine as-is.

```suggestion
  const [username, setUsername] = useState('');
```